packages/ui/src/ui/pages/query-tracker/QueryResultsVisualization/preparers/scatter.ts

```diff
@@ -0,0 +1,64 @@
+import type {ChartKitWidgetData} from '@gravity-ui/chartkit';
+import type {PrepareLineArgs} from './types';
+import {splitDataByColor} from './splitDataByColor';
+
+export function prepareScatter(args: PrepareLineArgs): ChartKitWidgetData {
+    const {visualization} = args;
+    const rows = args.result;
+    const xPlaceholder = visualization.placeholders.find((placeholder) => placeholder.id === 'x');
+    const yPlaceholder = visualization.placeholders.find((placeholder) => placeholder.id === 'y');
+    const colorPlaceholder = visualization.placeholders.find(
```

Comment:
It is better to make a reduce in which to find all these elements. This way we will reduce the number of passes through the same array